js-extensions/src/ClassMetadataStore.js

```diff
@@ -8,13 +8,20 @@ export class ClassMetadataStore {
          * Type info cache
          */
         this.classMetadata = {};
-        
+
         /**
          * Fetch function for the classMetadata
          */
         this.classMetadataProvider = classMetadataProvider;
+
+        /**
+         * Onload callbacks cache. Used to ensure we don't
+         * issue multiple in-parallel requests for the same
+         * class metadata.
+         */
+        this.classMetadataOnloadCallbacks = {};
```

Comment:
This is becoming too common of a pattern, we should generalize it at the `fetch` level for a given URL.